public/components/trace_analytics/components/traces/trace_table_helpers.tsx

```diff
@@ -73,28 +73,44 @@ export const getTableColumns = (
       '-'
     );
 
-  const renderTraceLinkField = (item: string) => (
-    <EuiFlexGroup gutterSize="s" alignItems="center">
-      <EuiFlexItem grow={false}>
-        <EuiLink
-          data-test-subj="trace-link"
-          {...(getTraceViewUri && { href: getTraceViewUri(item) })}
-          {...(openTraceFlyout && { onClick: () => openTraceFlyout(item) })}
-        >
-          <EuiText size="s" className="traces-table traces-table-trace-id" title={item}>
-            {item}
-          </EuiText>
-        </EuiLink>
-      </EuiFlexItem>
-      <EuiFlexItem grow={false}>
-        <EuiCopy textToCopy={item}>
-          {(copy) => (
-            <EuiButtonIcon aria-label="Copy trace id" iconType="copyClipboard" onClick={copy} />
-          )}
-        </EuiCopy>
-      </EuiFlexItem>
-    </EuiFlexGroup>
-  );
+  const renderTraceLinkField = (item: string) => {
+    // Extract the current mode from the URL or session storage
+    const queryParams = new URLSearchParams(window.location.search);
+    const traceMode = queryParams.get('mode') || sessionStorage.getItem('TraceAnalyticsMode');
+
+    // Update the href to include the mode if available
+    const updatedGetTraceViewUri = (traceId: string) => {
```

Comment:
"update get trace view uri" is not very clear, maybe
```suggestion
    const appendModeToTraceViewUri = (traceId: string) => {
```

Reply:
Updated, thank you.